src/shared/getExecutionResult.ts

```diff
@@ -1,146 +1,190 @@
-        ![ExecutionState.ExecutionSuccessful, ExecutionState.ExecutionFailed]
-            .includes(execution.state ?? ExecutionState.Nonexistent)
-    ) {
-        return null
+const getExecutedTransaction = async (
+  scheduledTxHash: string,
+  execution: Execution
+) => {
+  const provider = execution.provider as any;
+  const contractAddress = execution.config.contractAddress;
+  const window = execution.plan.window.toNumber();
+
+  const state = await execution.getState();
+
+  if (
+    ![
+      EExecutionState.ExecutionSuccessful,
+      EExecutionState.ExecutionFailed,
+    ].includes(state)
+  ) {
+    return null;
+  }
+
+  console.time("getExecutionResult");
+
+  const blocksProcessed: number[] = [];
+  const addProcess = (initialBlock: number, offset: number) => {
+    const processing = initialBlock + offset;
+    if (!blocksProcessed.includes(processing)) {
+      blocksProcessed.push(processing);
+      return getEventIfExist(provider, processing, contractAddress, execution);
     }
-    
-    console.time("getExecutionResult")
-
-    const blocksProcessed: number[] = []
-    const addProcess = (initialBlock: number, offset: number) => {
-        const processing = initialBlock + offset
-        if (!blocksProcessed.includes(processing)) {
-            blocksProcessed.push(processing)
-            return getEventIfExist(provider, processing, contractAddress, execution)
-        }
+  };
+
+  const executeAt = new Date(execution.executeAt);
+
+  const scheduledTx = await provider.getTransaction(scheduledTxHash);
+
+  const scheduledBlockNumber = scheduledTx.blockNumber!;
+  const lastBlockNumber = await provider.getBlockNumber();
+
+  const { blocksRate, fromDate: scheduledDate } = await getBlocksRateByMinute(
+    provider,
+    scheduledBlockNumber,
+    lastBlockNumber
+  );
+
+  const blockNumberExecuteAtMid = getEstimatedBlockNumber(
+    executeAt,
+    scheduledDate,
+    scheduledBlockNumber,
+    blocksRate
+  );
+  const blockNumberExecuteAtLow = getEstimatedBlockNumber(
+    subSeconds(executeAt, window),
+    scheduledDate,
+    scheduledBlockNumber,
+    blocksRate
+  );
+  const blockNumberExecuteAtUpper = getEstimatedBlockNumber(
+    addSeconds(executeAt, window),
+    scheduledDate,
+    scheduledBlockNumber,
+    blocksRate
+  );
+
+  const lowWindowBlock = Math.trunc(
+    Math.max(blockNumberExecuteAtLow, scheduledBlockNumber)
+  );
+  const upperWindowBlock = Math.trunc(
+    Math.min(blockNumberExecuteAtUpper, lastBlockNumber)
+  );
+  const midBlockNumber = Math.trunc(blockNumberExecuteAtMid);
+
+  let foundedEvent: IExecutedTransaction | null = null;
+  let counter = 0;
+  let running = [];
+  while (!foundedEvent) {
+    running.push(addProcess(lowWindowBlock, counter));
+
+    running.push(addProcess(midBlockNumber, -counter));
+
+    running.push(addProcess(midBlockNumber, counter));
+
+    running.push(addProcess(upperWindowBlock, -counter));
+
+    if (running.length === PROMISE_PARALLEL_NUMBER) {
```

Comment:
I don't like this condition. If PROMISE_PARALLEL_NUMBER is lower than 4, `foundedEvent` will never be set and it's and becomes an infinite loop. We should at least add a comment where the constant is initialized.